hutch_python/tests/test_ipython_session_timer.py

```diff
@@ -0,0 +1,52 @@
+import time
+import unittest
+
+import pytest
+from test_ipython_log import FakeIPython
+
+from hutch_python.ipython_session_timer import IPythonSessionTimer
+
+
+@pytest.fixture(scope='function')
+def fake_ipython():
+    fake_ipython = FakeIPython()
+    return fake_ipython
+
+
+@pytest.fixture(scope='function')
+def session_timer(fake_ipython):
+    session_timer = IPythonSessionTimer(fake_ipython)
+    session_timer.max_idle_time = 20.0
+    session_timer.user_active = True
+    session_timer.curr_time = time.monotonic()
+    session_timer.last_active_time = session_timer.curr_time - 10.0
+    session_timer.idle_time = session_timer.curr_time - session_timer.last_active_time
+    return session_timer
+
+
+def test_set_user_active(session_timer):
+    assert session_timer.user_active
+
+
+def test_set_user_inactive(session_timer):
+    session_timer.user_active = False
+    assert not session_timer.user_active
+
+
+def test_set_idle_time(session_timer):
+    assert session_timer.idle_time == 10.0
```

Comment:
I'm not entirely sure what these three tests are for, it seems to me like they're checking if the fixture is set up as expected.

Reply:
Oh, forgot about these. It's from when I didn't really understand how Pytest works. Fixed.